nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -568,164 +699,50 @@ public void verifyDeleteReportingTask(String reportingTaskId) {
-            .map(entry -> new Revision(entry.getValue().getVersion(), entry.getValue().getClientId(), entry.getKey()))
-            .forEach(rev -> requiredRevisions.add(rev));
-
-        snippetDto.getFunnels().entrySet().stream()
-            .map(entry -> new Revision(entry.getValue().getVersion(), entry.getValue().getClientId(), entry.getKey()))
-            .forEach(rev -> requiredRevisions.add(rev));
-
-        snippetDto.getInputPorts().entrySet().stream()
-            .map(entry -> new Revision(entry.getValue().getVersion(), entry.getValue().getClientId(), entry.getKey()))
-            .forEach(rev -> requiredRevisions.add(rev));
-
-        snippetDto.getOutputPorts().entrySet().stream()
-            .map(entry -> new Revision(entry.getValue().getVersion(), entry.getValue().getClientId(), entry.getKey()))
-            .forEach(rev -> requiredRevisions.add(rev));
-
-        snippetDto.getLabels().entrySet().stream()
-            .map(entry -> new Revision(entry.getValue().getVersion(), entry.getValue().getClientId(), entry.getKey()))
-            .forEach(rev -> requiredRevisions.add(rev));
-
-        snippetDto.getProcessors().entrySet().stream()
-            .map(entry -> new Revision(entry.getValue().getVersion(), entry.getValue().getClientId(), entry.getKey()))
-            .forEach(rev -> requiredRevisions.add(rev));
-
-        snippetDto.getRemoteProcessGroups().entrySet().stream()
-            .map(entry -> new Revision(entry.getValue().getVersion(), entry.getValue().getClientId(), entry.getKey()))
-            .forEach(rev -> requiredRevisions.add(rev));
-
-        for (final String groupId : snippetDto.getProcessGroups().keySet()) {
-            requiredRevisions.addAll(getRevisionsForGroup(groupId));
-        }
-
-        return requiredRevisions;
-    }
-
-    private ProcessGroup getGroup(final String groupId) {
-        return revisionManager.get(groupId, rev -> processGroupDAO.getProcessGroup(groupId));
-    }
-
     @Override
-    public UpdateResult<SnippetEntity> updateSnippet(final Revision revision, final SnippetDTO snippetDto) {
-        // if label does not exist, then create new label
-        if (snippetDAO.hasSnippet(snippetDto.getId()) == false) {
-            return new UpdateResult<>(createSnippet(snippetDto), true);
-        }
-
-        final Set<Revision> requiredRevisions = getRevisionsForSnippet(snippetDto);
-
-        // if the parent group is specified in the request, ensure write access to it as it could be moving the components in the snippet
-        final String requestProcessGroupIdentifier = snippetDto.getParentGroupId();
-        if (requestProcessGroupIdentifier != null) {
-            final ProcessGroup requestProcessGroup = processGroupDAO.getProcessGroup(requestProcessGroupIdentifier);
-            requestProcessGroup.authorize(authorizer, RequestAction.WRITE);
-        }
-
+    public SnippetEntity updateSnippet(final Set<Revision> revisions, final SnippetDTO snippetDto) {
```

Comment:
This name is a bit confusing, as it is not actually updating a snippet but rather updating the components referenced by the snippet. Perhaps we should rename it "updateSnippetComponents"?